Source/WebCore/platform/graphics/SourceBufferPrivate.cpp

```diff
@@ -695,26 +701,31 @@ Ref<MediaPromise> SourceBufferPrivate::append(Ref<SharedBuffer>&& buffer)
         return m_currentAppendProcessing->whenSettled(RunLoop::current(), [previousResult = WTFMove(result)](auto result) {
             return (previousResult && result) ? OperationPromise::createAndResolve() : OperationPromise::createAndReject(!result ? result.error() : previousResult.error());
         });
-    })->whenSettled(RunLoop::current(), [weakThis = WeakPtr { * this }, this, abortCount = m_abortCount](auto result) mutable -> Ref<OperationPromise> {
-        if (!weakThis || !result || !isAttached())
+    })->whenSettled(RunLoop::current(), [weakThis = ThreadSafeWeakPtr { * this }, this, abortCount = m_abortCount](auto result) mutable -> Ref<OperationPromise> {
+        RefPtr protectedThis = weakThis.get();
+        if (!protectedThis || !result)
             return OperationPromise::createAndReject(!result ? result.error() : PlatformMediaError::BufferRemoved);
         if (abortCount != m_abortCount)
             return OperationPromise::createAndResolve();
 
+        RefPtr client = this->client();
+        if (!client)
+            return OperationPromise::createAndReject(PlatformMediaError::BufferRemoved);
+
         // Resolve the changes in TrackBuffers' buffered ranges
         // into the SourceBuffer's buffered ranges
         auto trackBuffers = trackBuffersRanges();
-        client().sourceBufferPrivateTrackBuffersChanged(trackBuffers);
+        client->sourceBufferPrivateTrackBuffersChanged(trackBuffers);
 
         Vector<Ref<MediaPromise>> promises;
```

Comment:
It would be nice if Vector<Ref<MediaPromise>, 3> promises would work, this way no malloc needed.
Maybe MediaPromise::all could take an iterable.

Reply:
it probably will work.